### DISCUSSION TITLE
Partial rerender via raw window handle

### DISCUSSION THREAD

**Discussion by floers** [slint-ui/slint#6754] [Q&A]
I want to create a slint app where I can control which parts are rerendered (damaged). This is because it is a pen drawing app and I want e-ink devices to render only the pen strokes. I am aware this is not supported by slint directly atm (https://github.com/slint-ui/slint/discussions/3190) but shouldn't it be possible to use the raw window handle?

I render the pen strokes via opengl (`femtovg::renderer::opengl::OpenGL`) via `window.set_rendering_notifier`. Now instead of rerendering this on rerender request I thought I can do this on input event and then use wayland directly
```
// code that 
surface.damage_buffer(x,y,w,h);
surface.commit();
```

This doesn't work though and I think it is because slint did not change the underlying buffer unless `request_redraw` is called? But this also makes the compositor rerender the whole window. Could I get the surface buffer somehow and write my pen strokes directly inside? I am blocked by this and maybe on the wrong track. Is there any way I could accomplish my goal? If not: I would be happy to contribute to slint to make this possible but I would need some hints where and how this should be added...

**Comment by tronical** [comment]
I'm not sure if this is possible easily. But what you could do is to build from the bottom up. Make a new `slint::platform::Platform` and `WindowAdapter` by using winit directly and the `slint::platform::femtovg_renderer::FemtoVGRenderer`, and glutin. That way you have access to the GL context, and in your input event you could make it current on the window's surface, check the age of the back buffer (important!), and if it's suitable then you could try to do your shortcut there, swap buffers, and commit.

That said, partial rendering with FemtoVG is a little tricky in the sense that the scissoring is implemented in the fragment shader, so it will still run the whole pipeline even for invisible pixels, just `discard` in the fragment shader. It's correct, and anti-aliased, but it's not a way to avoid work.
